internal/services/storagemover/storage_mover_source_endpoint_resource_test.go

```diff
@@ -0,0 +1,180 @@
+	r := StorageMoverSourceEndpointTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStorageMoverSourceEndpoint_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_mover_source_endpoint", "test")
+	r := StorageMoverSourceEndpointTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccStorageMoverSourceEndpoint_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_mover_source_endpoint", "test")
+	r := StorageMoverSourceEndpointTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStorageMoverSourceEndpoint_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_mover_source_endpoint", "test")
+	r := StorageMoverSourceEndpointTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r StorageMoverSourceEndpointTestResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := endpoints.ParseEndpointID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	client := clients.StorageMover.EndpointsClient
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (r StorageMoverSourceEndpointTestResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-rg-%d"
+  location = "%s"
+}
+resource "azurerm_storage_mover" "test" {
+  name                = "acctest-ssm-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
+}
+
+func (r StorageMoverSourceEndpointTestResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+provider "azurerm" {
```

Comment:
Could this be moved into the template?

Reply:
Thanks for your feedback. Putting it outside of template because I got feedback from Tom a few days ago - "the provider block should be in each test, not the template - to allow for overriding it", see below for details. I assume that his feedback makes sense and I've been doing it ever since, WDYT?
![image](https://user-images.githubusercontent.com/39109137/232945921-f43fe275-3690-4e7a-bed1-3e415653c24e.png
)

Reply:
Ohhh good to note! Thank you!